### ISSUE TITLE
Add DEV_MODE Deprecation support

### ISSUE THREAD

**Issue by jasonLaster** [marionettejs/backbone.marionette#2760]
Marionette  `deprecate`  should support dev mode

```js
Marionette.DEV_MODE = true;
Marionette.deprecate('Composite View is now deprecated'); // console.warn Marionette: Composite View is now deprecated
Marionette.deprecate('Composite View is now deprecated'); // no-op
```

```js
Marionette.DEV_MODE = false;
Marionette.deprecate('Composite View is now deprecated'); //  no-op
```

NOTE: Backbone.Metal has great deprecation support so this is just a marionette wrapper with a DEV_MODE flag

**Comment by paulfalgout** [comment]
https://github.com/marionettejs/backbone.marionette/blob/next/src/utils/deprecate.js

**Comment by ahumphreys87** [comment]
fixed with #2766
